File: frontend/src/objects/domain/TicketKind.ts
export const TICKET_KIND = {
  positiveReview: 'PositiveReview',
  negativeReview: 'NegativeReview',
  deliveryIssue: 'DeliveryIssue',
} as const

export type TicketKind = (typeof TICKET_KIND)[keyof typeof TICKET_KIND]

File: backend/src/main/scala/services/admin/objects/TicketKind.scala
package services.admin.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import system.objects.*

import io.circe.{Decoder, Encoder}

enum TicketKind:
  case PositiveReview, NegativeReview, DeliveryIssue

object TicketKind:
  private val enumLabel = text("TicketKind")
  def render(value: TicketKind): DisplayText = enumDisplayText(value)
  def parse(value: DisplayText): Option[TicketKind] = parseEnumValue(value, TicketKind.values)
  given Encoder[TicketKind] = enumEncoder
  given Decoder[TicketKind] = enumDecoder(TicketKind.values, enumLabel)
